nxdrive/engine/dao/sqlite.py

```diff
@@ -2660,6 +2660,45 @@ def save_upload(self, upload: Upload) -> None:
             else:
                 self.transferUpdated.emit()
 
+    def save_dt_upload(self, upload: Upload) -> None:
+        """
+        New direct transfer upload.
+        Woll have the same status as it's Session on creation.
+        """
+        with self.lock:
+            # Remove non-serializable data, never used elsewhere
+            batch = {k: v for k, v in upload.batch.items() if k != "blobs"}
+
+            values = (
+                upload.path,
+                upload.doc_pair,
+                TransferStatus.CANCELLED.value,
```

Comment:
This should be the session status, right?

Reply:
Add a comment to ease understanding, this is way too magic combined with SQL `IFNULL()`.